### PR TITLE
[JENKINS-39062] Move the GitHubRepositoryDescription column to github branch source

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-39062](https://issues.jenkins-ci.org/browse/JENKINS-39062)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoAction.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import hudson.model.InvisibleAction;
+import java.net.URL;
+import jenkins.branch.MultiBranchProject;
+import org.kohsuke.github.GHRepository;
+
+/**
+ * Invisible property on {@link MultiBranchProject}
+ * that retains information about GitHub repository.
+ *
+ * @author Kohsuke Kawaguchi
```
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoAction.java]: See https://github.com/jenkinsci/github-organization-folder-plugin/pull/30 where this code was copied from... I'm basically removing the need for two separate plugins doing basically the same thing... when I am done, github-organization-folder-plugin will basically be just a pile of readResolve() methods and will be deprecated...

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoAction.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        this.description = Util.fixEmpty(description);
+        this.homepage = Util.fixEmpty(homepage);
+    }
+
+    public GitHubRepoAction(GitHubRepoAction that) {
+        this(that.getUrl(), that.getDescription(), that.getHomepage());
+    }
+
+    private Object readResolve() throws ObjectStreamException {
+        if ((description != null && StringUtils.isBlank(description))
```
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubRepoAction.java]: Basically, during data migration, there are some cases where a deserialized object will have an empty description where the description should be null... this readResolve will fix those such that the description is then correctly null for those cases

**Comment by rsandell** [comment]
:bee:

**Comment by stephenc** [comment]
@reviewbybees done

**Pull request merged**